File: 프로그래머스/lv2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    cnt = 0
    def dfs(idx, sum):
        nonlocal cnt
        if idx == len(numbers):
            if sum == target:
                cnt += 1
            return
        dfs(idx+1, sum + numbers[idx])
        dfs(idx+1, sum - numbers[idx])
    dfs(0,0)
    return cnt

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
    let cnt = 0
    const dfs = (depth,sum) => {
        if(sum===target && depth === numbers.length-1){
            cnt++;
            return
        }
        else if(depth === numbers.length-1) return
        dfs(depth+1,sum+numbers[depth+1])
        dfs(depth+1,sum-numbers[depth+1])
    }
    dfs(0,numbers[0])
    dfs(0,-numbers[0])
    return cnt
}